NuRadioMC/SignalProp/analyticraytracing.py

```diff
@@ -1718,11 +1750,448 @@ def get_path(self, iS, n_points=1000):
+            
+            dt = 1 / samp_rate * units.ns
+            
+            t_theta.set_trace(etheta, sampling_rate=samp_rate)
+            t_phi.set_trace(ephi, sampling_rate=samp_rate)
+
+        else:
+            self.__logger.error("error: wrong data type")
+
+        TT = t_theta.get_times()
+
+        shift = TT[t_theta.get_trace() == max(t_theta.get_trace())]
+
+        t_theta.set_trace_start_time(-shift)
+        t_phi.set_trace_start_time(-shift)
+
+        TT = t_theta.get_times()
+
+        time_delay_short = self.get_birefringence_time_delay(source, antenna, path_type=path_type, acc=acc)
+        polar_short = self.get_path_polarization(source, antenna, path_type=path_type, acc=acc)
+
+        polar_theta0 = polar_short[0][:, 1]
+        polar_phi0 = polar_short[0][:, 2]
+
+        polar_theta1 = polar_short[1][:, 1]
+        polar_phi1 = polar_short[1][:, 2]
+
+        diff = time_delay_short[2] - time_delay_short[1]
+
+        start_theta = t_theta.get_trace()
+        start_phi = t_phi.get_trace()
+
+        for i in range(len(diff)):
+            
+            a = polar_theta0[i]
+            b = polar_phi0[i]
+            
+            c = polar_theta1[i]
+            d = polar_phi1[i]
+
+            if np.isclose(a*d - b*c, 0) or np.isnan([a, b, c, d]).any():
+                self.__logger.warning("warning: Polarization vectors similar, R-matrix not invertible, iteration" + str(i))
+                
+                R = np.matrix([[1, 0], [0, 1]])
+                time_shift = 0
+            
+            else:
+                R = np.matrix([[a, b], [c, d]])
+                time_shift = diff[i]
+            
+            th = t_theta.get_frequency_spectrum()
+            ph = t_phi.get_frequency_spectrum()
+            t_slow.set_frequency_spectrum(th * R[0, 0] + ph * R[0, 1], sampling_rate=1 / dt)
+            t_fast.set_frequency_spectrum(th * R[1, 0] + ph * R[1, 1], sampling_rate=1 / dt)
+            
+            t_fast.apply_time_shift(time_shift)
+            
+            Rinv = np.linalg.inv(R)
+            
+            fa = t_fast.get_frequency_spectrum()
+            sl = t_slow.get_frequency_spectrum()
+            t_theta.set_frequency_spectrum(sl * Rinv[0, 0] + fa * Rinv[0, 1], sampling_rate=1 / dt)
+            t_phi.set_frequency_spectrum(sl * Rinv[1, 0] + fa * Rinv[1, 1], sampling_rate=1 / dt)
+        
+        end_theta = t_theta.get_trace()
+        end_phi = t_phi.get_trace()
+        
+        h_th = signal.hilbert(end_theta)
+        h_ph = signal.hilbert(end_phi)
+        
+        if style == 'single pulse':
+            t_delay = TT[h_th == max(h_th)] - TT[h_ph == max(h_ph)]
+            return(start_theta, start_phi, end_theta, end_phi, TT, t_delay)
+        
+        elif style == 'simulation pulse':
+            el_field = np.vstack((f_spectrum.get_frequency_spectrum()[0], t_theta.get_frequency_spectrum(), t_phi.get_frequency_spectrum()))
```

Comment:
confusing that you return the first value of `f_spectrum`. Better write `np.zeros(n_samples)` or similar.

Reply:
Unnecessary after the code was rewritten.